pkg/kubelet/kubelet_node_status.go

```diff
@@ -554,3 +568,56 @@ func validateNodeIP(nodeIP net.IP) error {
 	}
 	return fmt.Errorf("Node IP: %q not found in the host's network interfaces", nodeIP.String())
 }
+
+// nodeStatusHasChanged compares the original node and current node's status and
+// returns true if any change happens. The heartbeat timestamp is ignored.
+func nodeStatusHasChanged(originalStatus *v1.NodeStatus, status *v1.NodeStatus) bool {
+	if originalStatus == nil || status == nil {
+		return !apiequality.Semantic.DeepEqual(originalStatus, status)
```

Comment:
Why do you still need to compare here if one of them is nil? Am I missing something?...

Reply:
The full version is:
```
if originalStatus == nil && status == nil {
    return false
} else if originalStatus == nil && status != nil {
    return true
} else if originalStatus != nil && status == nil {
    return true
}
```
So we still need the comparison here if we use a shorter version of code.

Reply:
Done